lib/uds/upload_download.c

```diff
@@ -129,15 +180,51 @@ static UDSErr_t continue_download(const struct uds_context* const context) {
     return UDS_NRC_RequestOutOfRange;
   }
 
-  int rc = flash_write(flash_controller, upload_download_state.current_address,
-                       args->data, args->len);
+  LOG_INF("Writing to flash at addr 0x%08lx, size %u",
+          upload_download_state.current_address, args->len);
+
+  // Flash can only be written to in blocks of write_block_size, first we
+  // calculate the number of bytes that would overflow this block size.
+  // Then we write all full blocks first, and if there are remaining bytes, we
+  // create a temporary buffer to write the last block with padding (0xFF).
+  const size_t overflow_size =
+      args->len % upload_download_state.write_block_size;
+  const size_t first_write = args->len - overflow_size;
+
+  int rc;
+  if (first_write > 0) {
+    rc = flash_write(flash_controller, upload_download_state.current_address,
+                     args->data, first_write);
+
+    if (rc != 0) {
+      LOG_ERR("Flash write failed at addr 0x%08lx, size %u, err %d",
+              upload_download_state.current_address, first_write, rc);
+      return UDS_NRC_GeneralProgrammingFailure;
+    }
+  }
 
-  if (rc != 0) {
-    LOG_ERR("Flash write failed at addr 0x%08lx, size %u, err %d",
-            upload_download_state.current_address, args->len, rc);
-    return UDS_NRC_GeneralProgrammingFailure;
+  if (overflow_size != 0) {
+    // need to write remaining bytes
+    uint8_t last_bytes
```

Comment:
I thought about using `upload_download_state.write_block_size` as the size of this array. This would mean that the stack is allocated dynamically but would remove the necessity for the MAXIMUM macro